lib/avalanche/requests/statement_request.ex

```diff
@@ -67,9 +67,13 @@ defmodule Avalanche.StatementRequest do
   """
   def run(%__MODULE__{} = request, opts \\ []) do
     pipeline = build_pipeline(request, opts)
+    metadata = %{params: request.body.parameters, query: request.body.statement}
 
-    with {:ok, response} <- Req.Request.run(pipeline) do
-      handle_response(response)
+    with _ <- Avalanche.Telemetry.start(:query, metadata, %{}),
+         {:ok, response} <- Req.Request.run(pipeline),
+         response <- handle_response(response),
+         _ <- Avalanche.Telemetry.stop(:query, System.monotonic_time(), metadata, %{}) do
+      response
```

Comment:
Probably need to wrap the `with` with try/catch for exception or maybe move that all that way out to `Avalanche.run` https://github.com/HGInsights/avalanche/blob/main/lib/avalanche.ex#L162